File: android/ffmpeg-kit-android-lib/src/main/java/com/lgcns/ffmpegkit/LogCallback.java
package com.lgcns.ffmpegkit;

/**
 * <p>Callback function that receives logs generated for <code>FFmpegKit</code> sessions.
 */
@FunctionalInterface
public interface LogCallback {

    /**
     * <p>Called when a log entry is received.
     *
     * @param log log entry
     */
    void apply(final Log log);

}

File: android/ffmpeg-kit-android-lib/src/main/java/com/lgcns/ffmpegkit/LogCallback.kt
package com.lgcns.ffmpegkit

/**
 *
 * Callback function that receives logs generated for `FFmpegKit` sessions.
 */
fun interface LogCallback {
    /**
     *
     * Called when a log entry is received.
     *
     * @param log log entry
     */
    fun apply(log: Log?)
}